### ISSUE TITLE
Add "sameAs" to JSON-LD context so that object is a URI not a literal

### ISSUE THREAD

**Issue by rdmpage** [bionomia/bionomia#225]
## TL;DR

The `@context` for the JSON-LD output needs an entry for `sameAs`, e.g.

```
    "sameAs": {
      "@id": "sameAs",
      "@type": "@id"
    }
```

## Longer version

My understanding is that `sameAs` statements should connect two URIs. For example:

```
<https://bionomia.net/0000-0003-3522-9342> <http://schema.org/sameAs> <https://orcid.org/0000-0003-3522-9342> .
```

says that the person with URL `https://bionomia.net/0000-0003-3522-9342` is the same as the person with the ORCID `https://orcid.org/0000-0003-3522-9342`.

At the moment Bionomia JSON-LD statements such as `"sameAs": "https://orcid.org/0000-0003-3522-9342"` yield triples like:

```
<https://bionomia.net/0000-0003-3522-9342> <http://schema.org/sameAs> "https://orcid.org/0000-0003-3522-9342" .
```

Note that now ORCID id is a literal not a URI. This means anyone doing SPARQL queries will have to cast "https://orcid.org/0000-0003-3522-9342" to  `<https://orcid.org/0000-0003-3522-9342>` before they can link it to, for example, the RDF from ORCID.

The solution is to add `sameAs` to the `@context` (see above) so that when the JSON-LD is processed the object of the `sameAs` is treated as a URI. Alternatively you could use this uglier syntax each time you have a `sameAs` statement:

```
"sameAs" : { "@id" : "https://orcid.org/0000-0003-3522-9342" },
```

Putting it in the `@context` means you can still write nice simple URLs that are correctly interpreted.

This is one of several "gotchas" that have tripped me up over the last couple of years as I try to get my head around JSON-LD. 

Hope this makes sense. For now, when I process Bionomia JSON-LD I add the `sameAs` fragment so that my triples treat the ORCID as a URI.

**Comment by rdmpage** [comment]
Fair point. If you don't want to mess with the `@context` (which I confess I often do, mostly when I need to enforce a constraint that something is always an array, etc.) then doing it on each statement might be the way to go. For example, the current schema.org schema https://schema.org/version/latest/schemaorg-current-http.jsonld has items like this:

```
    {
      "@id": "schema:Thursday",
      "@type": "schema:DayOfWeek",
      "rdfs:comment": "The day of the week between Wednesday and Friday.",
      "rdfs:label": "Thursday",
      "schema:sameAs": {
        "@id": "http://www.wikidata.org/entity/Q129"
      }
    },
```

This makes it explicit that `http://www.wikidata.org/entity/Q129` is a URI. Slightly uglier, but less ambiguous. Of course, clients that don't care about RDF but just want JSON may go "huh?" when they see something like `{ "@id": "http://www.wikidata.org/entity/Q129" }` but this is the challenge. The cleanest JSON syntax has implications for the corresponding RDF, which in turn has implications for triple stores.

So I guess the options are:

1. leave it as it is and I monkey patch things at my end
2. you patch the context at your end
3. you write `sameAs` statements with `@id`
